Dockerfile

```diff
@@ -1,29 +1,35 @@
+FROM golang:1.17.2-bullseye as builder
+WORKDIR /go/src/app
+
+COPY ./builder/go.mod ./builder/go.sum ./
+RUN go mod download
+
+COPY ./builder .
+
+RUN go get -d -v ./...
+RUN go build .
+
+
 FROM node:slim as build
 WORKDIR /app
-ENV PATH /app/node_modules/.bin:$PATH
-COPY web/package.json ./
-COPY web/yarn.lock ./
-RUN yarn
-COPY web/ ./
-RUN yarn run build
-
-FROM golang:1.17.2-bullseye
-WORKDIR /go/src/app
 
 RUN apt-get update && apt-get install -y \
     poppler-utils \
     tesseract-ocr-spa \
     && rm -rf /var/lib/apt/lists/*
 
+ENV PATH /app/node_modules/.bin:$PATH
+COPY web/package.json ./
+COPY web/yarn.lock ./
+COPY web/ts/package.json ./ts/
+COPY web/ts/yarn.lock ./ts/
+RUN yarn
 
-COPY ./builder/go.mod ./builder/go.sum ./
-RUN go mod download
-
-COPY ./builder .
-
-RUN go get -d -v ./...
-RUN go install -v ./...
+WORKDIR /app/ts
+RUN yarn
```

Comment:
nitpick: here is where build-npm stops

Reply:
what do you mean?

Reply:
you could have split that in another container
- build-go
- build-npm
- run